### PR TITLE
Task/hw 61254 iOS add clear instance in core sdk, ui sdk and insights sdk

### PR DESCRIPTION
**Pull request by asenthil-altmtrk**

### CODE REVIEW

File: Cartfile [hunk 1]
```diff
@@ -1,2 +1,2 @@
-github "hyperwallet/hyperwallet-ios-sdk" "development"
-github "hyperwallet/hyperwallet-ios-insight" "1.0.0-beta01"
+github "hyperwallet/hyperwallet-ios-sdk" "task/HW-61254-iOS-Add-clearInstance-in-Core-SDK,-UI-SDK-and-Insights-SDK"
+github "hyperwallet/hyperwallet-ios-insight" "task/HW-61254-iOS-Add-clearInstance-in-Core-SDK,-UI-SDK-and-Insights-SDK"
```
Comment (jsaini-hw) [inline Cartfile]: change this back to development branches for both SDK and Insights once those PRs are merged in development branch

File: Common/Tests/Insights/HyperwalletInsightsTests.swift [hunk 2]
```diff
@@ -36,6 +36,7 @@ class HyperwalletInsightsTests: XCTestCase {
         if self.name.contains("testTrackError_ConfigNotInitialized") {
             return
         }
+        HyperwalletUI.clearInstance()
```
Comment (jsaini-hw) [inline Common/Tests/Insights/HyperwalletInsightsTests.swift]: Add separate func for clearInstance as suggested in Core SDK PR

File: Transfer/Sources/Extensions/HyperwalletTransferUI.swift [hunk 3]
```diff
@@ -18,10 +18,17 @@
...
+import TransferMethodRepository
+import TransferRepository
 #endif
 
 /// The HyperwalletUI extension
 public extension HyperwalletUI {
+    /// Clears TransferRepositoryFactory and TransferMethodRepositoryFactory instances.
+    static func transferUIClearInstance() {
+        TransferRepositoryFactory.clearInstance()
+        TransferMethodRepositoryFactory.clearInstance()
```
Comment (jsaini-hw) [inline Transfer/Sources/Extensions/HyperwalletTransferUI.swift]: UserRepositoryFactory.clearInstance()

File: TransferMethod/Sources/Extensions/HyperwalletTransferMethodUI.swift [hunk 4]
```diff
@@ -18,10 +18,15 @@
...
 #if !COCOAPODS
 import Common
+import TransferMethodRepository
 #endif
 
 /// The HyperwalletUI extension
 public extension HyperwalletUI {
+    /// Clears TransferMethodRepositoryFactory instance.
+    static func transferMethodUIClearInstance() {
+        TransferMethodRepositoryFactory.clearInstance()
```
Comment (jsaini-hw) [inline TransferMethod/Sources/Extensions/HyperwalletTransferMethodUI.swift]: UserRepositoryFactory.clearInstance()

Comment (asenthil-altmtrk) [reply TransferMethod/Sources/Extensions/HyperwalletTransferMethodUI.swift]: Added UserRepositoryFactory.clearInstance()

File: Transfer/Sources/Extensions/HyperwalletTransferUI.swift (hunk 3 above)
Comment (asenthil-altmtrk) [reply Transfer/Sources/Extensions/HyperwalletTransferUI.swift]: Added UserRepositoryFactory.clearInstance()

File: Common/Tests/Insights/HyperwalletInsightsTests.swift (hunk 2 above)
Comment (asenthil-altmtrk) [reply Common/Tests/Insights/HyperwalletInsightsTests.swift]: Added sperate testClearInstance

File: Common/Tests/Insights/HyperwalletInsightsTests.swift [hunk 5]
```diff
@@ -150,6 +151,19 @@ class HyperwalletInsightsTests: XCTestCase {
 
         XCTAssertNil(hyperwalletInsights?.insights, "Insights shouldn't be reloaded because of the wrong config")
     }
+
+    func testClearInstance() {
+        HyperwalletUI.setup(HyperwalletTestHelper.authenticationProvider)
+        let hyperwalletInstance1 = HyperwalletUI.shared
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply Common/Tests/Insights/HyperwalletInsightsTests.swift]: Renamed

File: Common/Tests/Insights/HyperwalletInsightsTests.swift [hunk 6]
```diff
@@ -150,6 +151,19 @@ class HyperwalletInsightsTests: XCTestCase {
...
+        XCTAssertNotNil(hyperwalletInstance1)
+        HyperwalletUI.clearInstance()
+        HyperwalletUI.setup(HyperwalletTestHelper.authenticationProvider)
+        let hyperwalletInstance2 = HyperwalletUI.shared
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply Common/Tests/Insights/HyperwalletInsightsTests.swift]: Renamed

File: Cartfile (hunk 1 above)
Comment (asenthil-altmtrk) [reply Cartfile]: Changed both to development

File: TransferMethod/Sources/Extensions/HyperwalletTransferMethodUI.swift [hunk 7]
```diff
@@ -43,9 +50,9 @@ public extension HyperwalletUI {
         -> SelectTransferMethodTypeCoordinator {
```
Comment (nirmal-altmtrk) [inline TransferMethod/Sources/Extensions/HyperwalletTransferMethodUI.swift]: Update this method as below.
Screenshot 2020-03-23 at 1 55 12 PM

File: Transfer/Sources/Extensions/HyperwalletTransferUI.swift [hunk 8]
```diff
@@ -18,10 +18,19 @@
...
 #endif
 
 /// The HyperwalletUI extension
 public extension HyperwalletUI {
+    /// Clears TransferRepositoryFactory , TransferMethodRepositoryFactory and UserRepositoryFactory instances.
+    static func transferUIClearInstance() {
+        TransferRepositoryFactory.clearInstance()
+        TransferMethodRepositoryFactory.clearInstance()
+        UserRepositoryFactory.clearInstance()
+    }
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply Transfer/Sources/Extensions/HyperwalletTransferUI.swift]: Added new line

File: TransferMethod/Sources/Extensions/HyperwalletTransferMethodUI.swift [hunk 9]
```diff
@@ -18,10 +18,17 @@
...
+import UserRepository
 #endif
 
 /// The HyperwalletUI extension
 public extension HyperwalletUI {
+    /// Clears TransferMethodRepositoryFactory and UserRepositoryFactory instances.
+    static func transferMethodUIClearInstance() {
+        TransferMethodRepositoryFactory.clearInstance()
+        UserRepositoryFactory.clearInstance()
+    }
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply TransferMethod/Sources/Extensions/HyperwalletTransferMethodUI.swift]: Added new line

File: TransferMethod/Sources/Extensions/HyperwalletTransferMethodUI.swift (hunk 7 above)
Comment (asenthil-altmtrk) [reply TransferMethod/Sources/Extensions/HyperwalletTransferMethodUI.swift]: Updated

**Pull request merged**